h2o-core/src/main/java/water/H2O.java

```diff
@@ -1965,6 +1992,49 @@ public void run() {
     }
   }
 
+  /**
+   * Register {@link NodeDisconnectHook}
+   * @param hook to register
+   */
+  public static void registerNodeDisconnectedHook(NodeDisconnectHook hook){
+    nodeDisconnectHooks.add(hook);
+  }
+
+  /**
+   *   Thread used to run disconnect hooks on nodes who disconnects from the cloud
+   */
+  public static class DisconnectedNodesWatchdog extends Thread {
```

Comment:
Oki, make sense. I just wanted to be consistent with rest of the classes ( seems like a lot of classes like this is there as well ), but totally agree with separating it to a new class file